### PR TITLE
push to add the thumbnail url for onedrive

### PR DESCRIPTION
**Pull request by darshan-patel**
@yogendra689 @ggerig 

1. Code to push thumbnail URL for one-drive and SharePoint.
2. Pushed deep link URL as Desktop URL for OneDrive and Mobile URL for SharePoint 

Kindly review 

Thanks

### CODE REVIEW

File: app/service/sharepoint_integration.rb [hunk 1]
```diff
@@ -109,4 +113,20 @@ def create_content_item(entry, parent_url)
 
     ContentItemCreationJob.perform_async(self.class.ecl_client_id, self.class.ecl_token, attributes)
   end
+
+  def thumbnail_url(record_id)
+    image_data = @sharepoint_communicator.files("/v1.0/drives/#{@drive_id}/items/#{record_id}/thumbnails")
+    response = []
+    if image_data && image_data["value"].any?
+      data = image_data["value"].first
+      # image_data["value"].each do |data|
```
Comment (yogendra689) [inline app/service/sharepoint_integration.rb]: remove unwanted code

File: app/service/sharepoint_integration.rb [hunk 2]
```diff
@@ -109,4 +113,20 @@ def create_content_item(entry, parent_url)
...
+        if(data && data['large'])
+          image_response = {}
+          image_response['url'] = data['large']['url']
```
Comment (yogendra689) [inline app/service/sharepoint_integration.rb]: if we can get more urls just pass them
data['small']['url'] and data['medium']['url']

File: app/service/sharepoint_integration.rb (hunk 1 above)
Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: removed unwanted code in https://github.com/edcast/collector_app/pull/48/commits/a6a08788ec35b27ededca3641b2f41a7d1b6d3fa

File: app/service/sharepoint_integration.rb (hunk 2 above)
Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: changed code to pass medium url in commit https://github.com/edcast/collector_app/pull/48/commits/a6a08788ec35b27ededca3641b2f41a7d1b6d3fa

**Comment by darshan-patel** [comment]
@yogendra689 
Moved deeplink genedration provider specific file
Kindly review

**Pull request merged**